tests/integration/security/mtls_cert_securenaming/mtls_cert_securenaming_test.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mtlscertsecurenaming
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/security/util"
+	"istio.io/istio/tests/integration/security/util/cert"
+	"istio.io/istio/tests/integration/security/util/connection"
+)
+
+const (
+	// The length of the example certificate chain.
+	exampleCertChainLength = 3
+
+	defaultIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/default"
+`
+	correctIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/b"
+`
+	nonExistIdentityDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "I-do-not-exist"
+`
+	identityListDR = `apiVersion: "networking.istio.io/v1alpha3"
+kind: "DestinationRule"
+metadata:
+  name: "service-b-dr"
+spec:
+  host: "b.NS.svc.cluster.local"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+      subjectAltNames:
+      - "spiffe://cluster.local/ns/NS/sa/a"
```

Comment:
the below is calling from a to b, why we add `sa/a` here, shouldn't it be only one service account? if you want to verify multiple service account work, we should add multiple deployment workload and each having different service account, sending request util all server endpoints have been reached.

Reply:
Correct. We could add that as well. I'll add a TODO here. We do also support this configuration, so I still like to keep it.

Reply:
Actually I'm a bit hesitate to further extend this test. That would definitely be a good one for E2E tests, but I don't think we want to complicate the integ tests too much.

Reply:
i still don't understand the current config. Why we want to add a list of unrelated random SAN here, as opposed to only have one `spiffe://cluster.local/ns/NS/sa/a`?  What we want to demonstrate with this config?

Reply:
It's just to mimic the scenario that there are multiple SAs running the service. I already have the other DR to cover the single SA case.